### ISSUE TITLE
may 2

### ISSUE THREAD

**Issue by hillarysanders** [hillarysanders/notes#24]
Yesterday:
- [x] slides
- [x] interview this morning
- [x]  make combined sophos, VT ROC curve. Compare FPR rate in VT (training dataset, validation set) data to FPR rate in Sophos benign data on final fitted convnet.

Took a lot longer than expected because doing it the brute force way would have taken hours, and sklearn didn't have a utility for it, so I had to dive in deep and write one on my own. *Still need to functionize it, but I'll add it to the utils repo!*

Also took longer because I had to develop a utility to test arbitrary models on arbitrary DBs. Before it would add a column to the db but for more one-off testing I wanted a different output.

TODO:
- [ ] encoding of URL characters: look at very similar URLs (sophos, VT) - are they different or the same? Due to encoding differences?
- [ ] train on data up to some timestamp, test on data observed in the next month.  measure ROC in two cases: case 1) allow observations to appear in the test data even if we saw them in the training data as long as they are observed after the timestamp and 2) just look at observations newly observed after the timestamp.  in both cases, weight the ROC curves by observation frequency.  in both cases, in the test data, use sophos customer data.

- html file. lstm model.

**Comment by hillarysanders** [comment]
# todo:
# todo:
# todo:
# todo:
# todo:
# 1) port this stuff to utils() branch, change neccessary dependencies. Do this while data thing is running.
# 2) write a little program to modify sqlite db train / testing values based on date. Kick that off on the deep learn ec2 instance.
# 3) kick off a heavier weight data fetch, after deleting data on current fetch ec2 instance?
